classes/class-wc-connect-shipping-label.php

```diff
@@ -0,0 +1,310 @@
+		}
+
+		protected function get_form_layout() {
+			$layout = [];
+
+			$address_fields = array(
+				array(
+					'key' => 'first_name',
+				),
+				array(
+					'key' => 'last_name',
+				),
+				array(
+					'key' => 'company',
+				),
+				array(
+					'key' => 'address_1',
+				),
+				array(
+					'key' => 'address_2',
+				),
+				array(
+					'key' => 'city',
+				),
+				array(
+					'key' => 'state',
+				),
+				array(
+					'key' => 'postcode',
+				),
+				array(
+					'key' => 'country',
+				),
+			);
+			$address_summary = '{first_name} {last_name}\\n{address_1} {address_2}\\n{city}, {postcode} {state}, {country}';
+
+			$orig_address_fields = $address_fields;
+			$dest_address_fields = $address_fields;
+			foreach( $address_fields as $index => $_ ) {
+				$orig_address_fields[ $index ][ 'key' ] = 'orig_' . $address_fields[ $index ][ 'key' ];
+				$dest_address_fields[ $index ][ 'key' ] = 'dest_' . $address_fields[ $index ][ 'key' ];
+			}
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Origin',
+				'title' => 'Generate shipping label: Address verification (origin)',
+				'description' => 'asdfghjkjhgfdsaASDFGHJJHGFDSA',
+				'items' => array( array( 'items' => $orig_address_fields ) ),
+				'bypass_suggestion_flag' => 'orig_bypass_suggestion',
+				'summary' => str_replace( '{', '{orig_', $address_summary ),
+				'sugggestion_hint' => 'vnkdjfitysncgkgse,jhxccvnaluerghsldjfvbzdj',
+			);
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Destination',
+				'title' => 'Generate shipping label: Address verification',
+				'description' => 'asdfghjkjhgfdsaASDFGHJJHGFDSA',
+				'items' => array( array( 'items' => $dest_address_fields ) ),
+				'bypass_suggestion_flag' => 'dest_bypass_suggestion',
+				'summary' => str_replace( '{', '{dest_', $address_summary ),
+				'sugggestion_hint' => 'qwertyuiopqwertyuiopqwerty uiopqwertyuiopqwertyuiop',
+			);
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Packages',
+				'title' => 'Shopping cart contents',
+				'items' => array(
+					array(
+						'items' => array(
+							array(
+								'key' => 'cart',
+								'type' => 'shopping_cart',
+							),
+						),
+					),
+				),
+				'summary' => '{cart}',
+			);
+
+			$layout[] = array(
+				'type' => 'step',
+				'tab_title' => 'Rates',
+				'title' => 'Select a shipping service',
+				'items' => array(
+					array(
+						'items' => array(
+							array(
+								'key' => 'rate',
+								'type' => 'radios',
+								'titleMap' => array(
+									'media' => 'Media Mail ($1.00)',
+									'pri_1' => 'Priority 1-day ($9.99)',
+								),
+							),
+						),
+					),
+				),
+				'summary' => '{rate}',
+			);
+
+			$layout[] = array(
+				'type' => 'summary',
+				'tab_title' => 'Preview',
+				'title' => 'dfhdfghdfgh',
+				'action_label' => 'Print',
+				'confirmation_flag' => 'confirm',
+			);
+
+			return $layout;
+		}
+
+		protected function get_form_schema() {
```

Comment:
The idea with the schema is that the Hydra server will receive a plain array (not nested). So, for example, after the first step, the server will receive all the `orig_address`, `orig_city`, `orig_whatever` fields filled, and the rest of the fields blank. In the server-side validation, it will return something for the origin address' individual fields (`OK`, `ERROR`, `suggestion`), and simply return errors for the rest of the fields.

So, the idea is that the client will send the entire form, and the server will short-circuit validation when appropiate.